dallinger/command_line.py

```diff
@@ -318,6 +319,20 @@ def deploy_sandbox_shared_setup(verbose=True, app=None, web_procs=1, exp_config=
     (id, tmp) = setup_experiment(debug=False, verbose=verbose, app=app,
                                  exp_config=exp_config)
 
+    # Register the experiment on OSF
+    # Load configuration.
+    config = get_config()
+    if not config.ready:
+        config.load_config()
+
+    try:
+        config.get('osf_access_token')
+        log("Preregistering experiment on OSF...")
+        registration.register(id, snapshot=None)
+
+    except Exception:
+        pass
```

Comment:
That's an improvement. I would probably write:
```python
if config.get('osf_access_token', None) is not None:
    config.get("osf_access_token")
    osf_id = _create_osf_project(dlgr_id)
    _upload_assets_to_OSF(dlgr_id, osf_id)
```
or
```python
try:
    config.get("osf_access_token")
except KeyError:
    pass
else:
    osf_id = _create_osf_project(dlgr_id)
    _upload_assets_to_OSF(dlgr_id, osf_id)
```
so that we don't accidentally catch KeyErrors that might happen in the other two functions that are called.